hr_holidays_public/models/hr_leave.py

```diff
@@ -8,6 +8,13 @@
 class HrLeave(models.Model):
     _inherit = "hr.leave"
 
+    def action_validate(self):
+        if self.holiday_status_id.exclude_public_holidays or not self.holiday_status_id:
```

Comment:
Put a docstring like this one for knowing the purpose of this overriding, or in the future, it may be removed if its goal is not understood:
```
"""Inject the needed context for excluding public holidays (if applicable) on the
actions derived from this validation. This is required for example for
`project_timesheet_holidays` for not generating the timesheet on the public holiday.
Unfortunately, no regression test can be added, being in a separate module.
"""
```

Reply:
Docstring added